pkg/controllers/cloud/networkpolicy.go

```diff
@@ -860,13 +870,13 @@ func (a *appliedToSecurityGroup) updateRules(r *NetworkPolicyReconciler) error {
 	if a.retryOp != nil {
 		return nil
 	}
-	nps, err := r.networkPolicyIndexer.ByIndex(networkPolicyIndexerByAppliedToGrp, a.id.Name)
+	nps, err := r.networkPolicyIndexer.ByIndex(networkPolicyIndexerByAppliedToGrp, a.id.Name.Name)
 	if err != nil {
-		return fmt.Errorf("unable to get networkPolicy with key %s from indexer: %w", a.id.Name, err)
+		return fmt.Errorf("unable to get networkPolicy with key %s from indexer: %w", a.id.Name.Name, err)
```

Comment:
Name.Name is not good. Consider declaring 
type CloudResource struct {
	Type CloudResourceType
	Name CloudResourceID
}

as 
type CloudResource struct {
	Type CloudResourceType
	CloudResourceID
}

In this way, you can directly access Name and VPC from Cloud Resource Struct

Reply:
There are a lot of functions which expect CloudResourceID(which mean we pass cloudresource.Name from calling functions).
We will have to change all those functions to have CloudResource instead of CloudResourceID. Also, the callers, it cascades.
Also, there is GetCloudName function which is referenced in many places, we need to change all calling functions. How about changing Name to Id ? As we its of type CloudResourceId.